### ISSUE TITLE
508-defect-2 [COGNITION]: Error message MUST be specific to the field

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#16784]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## User Story or Problem Statement

As a user, I want to know what exactly is wrong with my inputs so that I can fix any mistakes.

## Details

On Step 1: Applicant Information, the "city" input is updated to a new field APO/FPO/DPO upon selecting the military base checkbox. If an error is already flagged with the city input, it will persist when changed to APO/FPO/DPO. This results in an error message that is not specific to the input it is referring to.

## Acceptance Criteria

- [ ] Update the error message from city to APO/FPO/DPO.

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: production, staging

## Steps to Recreate

1. Reach Step 1: Applicant Information in VA Form 28-8832.
2. Do not fill out any fields and click continue.
3. Confirm the error message "City is required" shows for the city input.
4. Select the "I live on a..." checkbox.
5. Verify the "City is required" error message persists on the page.

## Solution 

Ensure the error message updates to a APO/FPO/DPO specific message. For example, "APO/FPO/DPO selection required."

## WCAG or Vendor Guidance 

* [WCAG 2.0: 3.3.1: Error Identification A](https://www.w3.org/WAI/WCAG21/Understanding/error-identification)
The intent of this Success Criterion is to ensure that users are aware that an error has occurred and can determine what is wrong. The error message should be as specific as possible.

## Screenshots or Trace Logs

<img width="521" alt="Screen Shot 2020-12-01 at 5 33 25 PM" src="https://user-images.githubusercontent.com/14154792/100804696-52916400-33fb-11eb-85b5-8eda39bf9fba.png">

**Comment by ohgreatkate** [comment]
After selecting the checkbox for living on a military base outside the US the error message does not change. The error message when checked should say "APO/FPO/DPO selection required."


![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/77993286/6e3ae049-66fc-409b-9bf9-a913ea468ffe)

**Comment by tblackwe** [comment]
Assigned to Fred

**Comment by fred-khoury-oddball** [comment]
@tblackwe  confirmed this issue exists and can be reproduced

**Comment by zack** [comment]
@Mottie @saderagsdale Is there any update on this being resolved in the forms library? This was assigned to our team but given existing discussion on this ticket I'm not sure why.

**Comment by saderagsdale** [comment]
@zack Please ask @steele-lm, as she reassigned it.

**Comment by jeana-adhoc** [comment]
Forms team has taken this on and will [address the error message on the "City"](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1228) when a military address is selected in the forms library.
